destination/destination.go

```diff
@@ -0,0 +1,128 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
```

Comment:
I just noticed this - I suggest putting the destination in the root of the module, it will read better (`grpcclient.Destination` instead of `destination.Destination`), also it will be easier to share any common code between the source and destination (currently we'd have to put that code into yet another package). If some part of the destination gets too complicated, it can be moved into a separate package, but I think it would be a nice pattern if all destinations provide the main structs in the root of the module (`Connector`, `Destination`, `Source` and `Specification`).